sign1.go

```diff
@@ -80,16 +86,22 @@ func (m *Sign1Message) UnmarshalCBOR(data []byte) error {
 		return errors.New("cbor: UnmarshalCBOR on nil Sign1Message pointer")
 	}
 
-	// fast message check
-	if !bytes.HasPrefix(data, sign1MessagePrefix) {
-		return errors.New("cbor: invalid COSE_Sign1_Tagged object")
-	}
-
 	// decode to sign1Message and parse
 	var raw sign1Message
-	if err := decModeWithTagsForbidden.Unmarshal(data[1:], &raw); err != nil {
+	var err error
+	var isUntagged bool
+
+	if bytes.HasPrefix(data, sign1MessagePrefix) {
+		err = decModeWithTagsForbidden.Unmarshal(data[1:], &raw)
+		isUntagged = false
+	} else {
+		err = decModeWithTagsForbidden.Unmarshal(data, &raw)
+		isUntagged = true
+	}
+	if err != nil {
 		return err
 	}
```

Comment:
We still can perform the fast message check

```suggestion
	// fast message check
	isUntagged := !bytes.HasPrefix(data, sign1MessagePrefix)
	if isUntagged {
		if !bytes.HasPrefix(data, sign1MessagePrefix[1:]) {
			return errors.New("cbor: invalid COSE_Sign1 / COSE_Sign1_Tagged object")
		}
		data = data[1:]
	}

	// decode to sign1Message and parse
	if err := decModeWithTagsForbidden.Unmarshal(data, &raw); err != nil {
		return err
	}
```